editor/plugin-slots/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Fill, withHooks } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+
+class PluginSlots extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.registerFill = this.registerFill.bind( this );
+
+		this.state = {
+			fills: {},
+		};
+	}
+
+	componentDidMount() {
+		this.props.hooks.addAction( 'registerFill', 'plugin-slots-register', this.registerFill );
```

Comment:
Do we really need to update the fills after the initial render? I'm thinking the hooks could be global and just provide them in the editorSettings object. Could simplify things and avoid another provider.

Reply:
>Do we really need to update the fills after the initial render? I'm thinking the hooks could be global and just provide them in the editorSettings object. Could simplify things and avoid another provider.

This could work, and is similar to the second "alternative" option described in the original comment (with some downsides mentioned).